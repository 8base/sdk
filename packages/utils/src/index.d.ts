
export type FieldSchema = {
  id: string,
  name: string,
  displayName?: string,
  description?: string,
  fieldType: string,
  fieldTypeAttributes: Object,
  isSystem: boolean,
  isList: boolean,
  isMeta: boolean,
  isRequired: boolean,
  isExternal: boolean,
  isUnique?: boolean,
  defaultValue: any,
  relation: {
    id: string,
    relationTableName: string,
    relationFieldName: string,
    refFieldName: string,
    refFieldDisplayName: string,
    refFieldIsList: boolean,
    refFieldIsRequired: boolean,
    refTable: {
      id: string,
    },
  },
};

export type TableSchema = {
  id: string,
  name: string,
  displayName?: string,
  isSystem?: boolean,
  fields: Array<FieldSchema>,
};

export type QueryGeneratorConfig = {
  deep?: number,
  withMeta?: boolean,
  relationItemsCount?: number,
  withResultData?: boolean,
  includeColumns?: null | string[],
};

type CreateQueryColumnsListConfig = {
  flatten?: boolean,
} & QueryGeneratorConfig


export type Schema = Array<TableSchema>;

export function createQueryColumnsList(
  tablesList: TableSchema[],
  tableName: string,
  config?: CreateQueryColumnsListConfig,
): Array<{ name: string, title: string, meta: Object }>;


export const FIELD_TYPE: {
  DATE: 'DATE',
  FILE: 'FILE',
  ID: 'ID',
  NUMBER: 'NUMBER',
  RELATION: 'RELATION',
  SWITCH: 'SWITCH',
  TEXT: 'TEXT',
  SMART: 'SMART',
  JSON: 'JSON',
};

export const SWITCH_FORMATS: {
  ON_OFF: 'ON_OFF',
  YES_NO: 'YES_NO',
  TRUE_FALSE: 'TRUE_FALSE',
  ACTIVE_INACTIVE: 'ACTIVE_INACTIVE',
  HIGH_LOW: 'HIGH_LOW',
  CUSTOM: 'CUSTOM',
};

export const SWITCH_VALUES: {
  ON_OFF: {
    true: 'On',
    false: 'Off',
  },
  YES_NO: {
    true: 'Yes',
    false: 'No',
  },
  TRUE_FALSE: {
    true: 'True',
    false: 'False',
  },
  ACTIVE_INACTIVE: {
    true: 'Active',
    false: 'Inactive',
  },
  HIGH_LOW: {
    true: 'High',
    false: 'Low',
  },
};

export const TEXT_FORMATS: {
  UNFORMATTED: 'UNFORMATTED',
  NAME: 'NAME',
  SSN: 'SSN',
  EIN: 'EIN',
  PHONE: 'PHONE',
  EMAIL: 'EMAIL',
};

export const NUMBER_FORMATS: {
  NUMBER: 'NUMBER',
  CURRENCY: 'CURRENCY',
  PERCENTAGE: 'PERCENTAGE',
  FRACTION: 'FRACTION',
  SCIENTIFIC: 'SCIENTIFIC',
};

export const FILE_FORMATS: {
  FILE: 'FILE',
  IMAGE: 'IMAGE',
};

export const SMART_FORMATS: {
  ADDRESS: 'ADDRESS',
  PHONE: 'PHONE',
};

export const DATE_FORMATS: {
  DATE: 'DATE',
  DATETIME: 'DATETIME',
};

export const MUTATION_TYPE: {
  CREATE: 'CREATE',
  UPDATE: 'UPDATE',
};

export const SYSTEM_TABLES: {
  USERS: 'Users',
  FILES: 'Files',
  SETTINGS: 'Settings',
  ROLES: 'Roles',
  INVITATIONS: 'Invitations',
  PERMISSIONS: 'Permissions',
  API_TOKENS: 'ApiTokens',
  ENVIRONMENT_VARIABLES: 'EnvironmentVariables',
};

export const MUTATION_FILE_FIELDS: ['fileId', 'public', 'filename', 'meta', 'mods'];

export type QueryGeneratorData = {
  tablesList: TableSchema[],
  tableName: string,
  appName?: string,
};

export const tablesListSelectors: {
  [key: string]: Function
}

export const tableSelectors: {
  [key: string]: Function
}

export const tableFieldSelectors: { [key: string]: Function };

export const throwIfMissingRequiredOption: Function;

export type AuthState = {
  email?: string,
  userId?: string,
  workspaceId?: string,
  refreshToken?: string,
  token?: string,
}

export type AuthData = {
  isEmailVerified: boolean,
  idToken: string,
  email: string,
  idTokenPayload: Object,
  state: any,
};

export interface AuthClient {
  getAuthState(): Promise<AuthState>,
  setAuthState(state: AuthState): Promise<void>,
  purgeAuthState(): Promise<void>,
  checkIsAuthorized(): Promise<boolean>,
}

export interface Authorizable {
  authorize(options?: Object): Promise<AuthData>,
  renewToken(options?: Object): Promise<AuthData>,
  changePassword(): Promise<{ email: string }>,
  logout(options?: Object): Promise<void>,
}
