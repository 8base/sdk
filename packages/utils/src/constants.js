//@flow

export const FIELD_TYPE = {
  DATE: 'DATE',
  FILE: 'FILE',
  ID: 'ID',
  NUMBER: 'NUMBER',
  RELATION: 'RELATION',
  SWITCH: 'SWITCH',
  TEXT: 'TEXT',
  SMART: 'SMART',
};

export const SWITCH_FORMATS = {
  ON_OFF: 'ON_OFF',
  YES_NO: 'YES_NO',
  TRUE_FALSE: 'TRUE_FALSE',
  ACTIVE_INACTIVE: 'ACTIVE_INACTIVE',
  HIGH_LOW: 'HIGH_LOW',
  SMART: 'SMART',
};

export const TEXT_FORMATS = {
  UNFORMATTED: 'UNFORMATTED',
  NAME: 'NAME',
  ADDRESS: 'ADDRESS',
  SSN: 'SSN',
  EIN: 'EIN',
  PHONE: 'PHONE',
  EMAIL: 'EMAIL',
};

export const NUMBER_FORMATS = {
  NUMBER: 'NUMBER',
  CURRENCY: 'CURRENCY',
  PERCENTAGE: 'PERCENTAGE',
  FRACTION: 'FRACTION',
  SCIENTIFIC: 'SCIENTIFIC',
};

export const FILE_FORMATS = {
  FILE: 'FILE',
  IMAGE: 'IMAGE',
};

export const DATE_FORMATS = {
  DATE: 'DATE',
  DATETIME: 'DATE/TIME',
};

export const MUTATION_TYPE = {
  CREATE: 'CREATE',
  UPDATE: 'UPDATE',
};
