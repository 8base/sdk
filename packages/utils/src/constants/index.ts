// @flow

export * from './schemaConstants';
export * from './countryCodes';
