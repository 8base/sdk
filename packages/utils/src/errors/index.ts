export { SDKError } from './SDKError';
export { ERROR_CODES } from './codes';

