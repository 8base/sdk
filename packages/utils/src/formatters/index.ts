export { formatDataForMutation } from './formatDataForMutation';
export { formatDataAfterQuery } from './formatDataAfterQuery';
