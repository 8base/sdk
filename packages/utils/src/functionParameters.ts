// @flow
import * as R from 'ramda';

const hasOption: (string | Array<string>) => ({}) => boolean = R.ifElse(
  R.is(String),
  R.has,
  R.hasPath,
);

const stringifyPath: (string | Array<string>) => string = R.ifElse(
  R.is(String),
  (path) => path,
  R.join('.'),
);

export const throwIfMissing = (paramName: string): void => {
  throw new Error(`Missing parameter: ${paramName}`);
};

export const throwIfMissingRequiredOption = (
  requiredOptionPaths: Array<Array<string> | string>,
  options: {},
): void => {
  requiredOptionPaths.forEach(optionPath => {
    const isMissing = !hasOption(optionPath)(options);

    if (isMissing) {
      throw new Error(`Missing option: ${stringifyPath(optionPath)}`);
    }
  });
};

