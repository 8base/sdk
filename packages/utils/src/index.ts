import * as tableSelectors from './selectors/tableSelectors';

export * from './constants';
export * from './formatters';
export * from './selectors';
export * from './verifiers';
export * from './queryGenerators';
export * from './functionParameters';
export * from './errors';

export { tableSelectors };
