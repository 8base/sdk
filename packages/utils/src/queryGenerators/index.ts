// @flow

export * from './queryGenerators';
export { createQueryColumnsList } from './createQueryColumnsList';
