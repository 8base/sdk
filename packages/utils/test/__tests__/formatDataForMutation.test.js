//@flow

import { formatDataForMutation, MUTATION_TYPE } from '../../src';

const SCHEMA = [{
  id: 'TABLE_SCHEMA_ID',
  name: 'tableSchema',
  displayName: 'Table Schema',
  isSystem: false,
  fields: [
    {
      name: 'scalar',
      displayName: 'Scalar',
      description: null,
      fieldType: 'TEXT',
      fieldTypeAttributes: {
        format: 'UNFORMATTED',
        fieldSize: 100,
      },
      isList: false,
      isRequired: false,
      isUnique: false,
      defaultValue: 'Scalar Default Value',
      relation: null,
    },
    {
      name: 'file',
      displayName: 'File',
      description: null,
      fieldType: 'FILE',
      fieldTypeAttributes: {
        format: 'FILE',
        showTitle: null,
        showUrl: null,
        maxSize: null,
        typeRestrictions: null,
      },
      isList: false,
      isRequired: false,
      isUnique: null,
      defaultValue: null,
      isSystem: false,
      isMeta: false,
      relation: null,
    },
    {
      name: 'fileList',
      displayName: 'File List',
      description: null,
      fieldType: 'FILE',
      fieldTypeAttributes: {
        format: 'FILE',
        showTitle: null,
        showUrl: null,
        maxSize: null,
        typeRestrictions: null,
      },
      isList: true,
      isRequired: false,
      isUnique: null,
      defaultValue: null,
      isSystem: false,
      isMeta: false,
      relation: null,
    },
    {
      name: 'scalarList',
      displayName: 'Scalar List',
      description: null,
      fieldType: 'TEXT',
      fieldTypeAttributes: {
        format: 'UNFORMATTED',
        fieldSize: 100,
      },
      isList: true,
      isRequired: false,
      isUnique: false,
      defaultValue: 'Scalar List Default Value 1',
      relation: null,
    },
    {
      name: 'relation',
      displayName: 'Relation',
      description: null,
      fieldType: 'RELATION',
      fieldTypeAttributes: null,
      isList: false,
      isRequired: false,
      isUnique: null,
      defaultValue: null,
      relation: {
        id: 'RELATION_FIELD_ID_1',
        relationTableName: 'RELATION_TABLE_NAME_1',
        relationFieldName: 'aid',
        refTable: {
          id: 'RELATION_TABLE_SCHEMA_ID',
        },
        refFieldIsList: false,
        refFieldIsRequired: true,
      },
    },
    {
      name: 'relationList',
      displayName: 'RelationList',
      description: null,
      fieldType: 'RELATION',
      fieldTypeAttributes: null,
      isList: true,
      isRequired: false,
      isUnique: null,
      defaultValue: null,
      relation: {
        id: 'RELATION_FIELD_ID_2',
        relationTableName: 'RELATION_TABLE_NAME_2',
        relationFieldName: 'aid',
        refTable: {
          id: 'RELATION_TABLE_SCHEMA_ID',
        },
        refFieldIsList: false,
        refFieldIsRequired: true,
      },
    },
  ],
}, {
  id: 'RELATION_TABLE_SCHEMA_ID',
  name: 'relationTableSchema',
  displayName: 'Relation Table Schema',
  isSystem: false,
  fields: [
    {
      name: 'scalar',
      displayName: 'Scalar',
      description: null,
      fieldType: 'TEXT',
      fieldTypeAttributes: {
        format: 'UNFORMATTED',
        fieldSize: 100,
      },
      isList: false,
      isRequired: false,
      isUnique: false,
      defaultValue: 'Scalar Default Value',
      relation: null,
    },
    {
      name: 'scalarList',
      displayName: 'Scalar List',
      description: null,
      fieldType: 'TEXT',
      fieldTypeAttributes: {
        format: 'UNFORMATTED',
        fieldSize: 100,
      },
      isList: true,
      isRequired: false,
      isUnique: false,
      defaultValue: 'Scalar List Default Value 1',
      relation: null,
    },
    {
      name: 'relation',
      displayName: 'Relation',
      description: null,
      fieldType: 'RELATION',
      fieldTypeAttributes: null,
      isList: false,
      isRequired: false,
      isUnique: null,
      defaultValue: null,
      relation: {
        id: 'RELATION_FIELD_ID_3',
        relationTableName: 'RELATION_TABLE_NAME_1',
        relationFieldName: 'aid',
        refTable: {
          id: 'TABLE_SCHEMA_ID',
        },
        refFieldIsList: false,
        refFieldIsRequired: true,
      },
    },
    {
      name: 'relationList',
      displayName: 'RelationList',
      description: null,
      fieldType: 'RELATION',
      fieldTypeAttributes: null,
      isList: true,
      isRequired: false,
      isUnique: null,
      defaultValue: null,
      relation: {
        id: 'RELATION_FIELD_ID_4',
        relationTableName: 'RELATION_TABLE_NAME_2',
        relationFieldName: 'aid',
        refTable: {
          id: 'TABLE_SCHEMA_ID',
        },
        refFieldIsList: false,
        refFieldIsRequired: true,
      },
    },
    {
      name: 'nestedRelation',
      displayName: 'Nested Relation',
      description: null,
      fieldType: 'RELATION',
      fieldTypeAttributes: null,
      isList: false,
      isRequired: false,
      isUnique: null,
      defaultValue: null,
      relation: {
        id: 'RELATION_FIELD_ID_5',
        relationTableName: 'RELATION_TABLE_NAME_3',
        relationFieldName: 'aid',
        refTable: {
          id: 'NESTED_RELATION_TABLE_SCHEMA_ID',
        },
        refFieldIsList: false,
        refFieldIsRequired: true,
      },
    },
    {
      name: 'nestedRelationList',
      displayName: 'Nested Relation List',
      description: null,
      fieldType: 'RELATION',
      fieldTypeAttributes: null,
      isList: true,
      isRequired: false,
      isUnique: null,
      defaultValue: null,
      relation: {
        id: 'RELATION_FIELD_ID_6',
        relationTableName: 'RELATION_TABLE_NAME_4',
        relationFieldName: 'aid',
        refTable: {
          id: 'NESTED_RELATION_TABLE_SCHEMA_ID',
        },
        refFieldIsList: false,
        refFieldIsRequired: true,
      },
    },
  ],
}, {
  id: 'NESTED_RELATION_TABLE_SCHEMA_ID',
  name: 'nestedRelationTableSchema',
  displayName: 'Other Relation Table Schema',
  isSystem: false,
  fields: [
    {
      name: 'scalar',
      displayName: 'Scalar',
      description: null,
      fieldType: 'TEXT',
      fieldTypeAttributes: {
        format: 'UNFORMATTED',
        fieldSize: 100,
      },
      isList: false,
      isRequired: false,
      isUnique: false,
      defaultValue: 'Scalar Default Value',
      relation: null,
    },
    {
      name: 'scalarList',
      displayName: 'Scalar List',
      description: null,
      fieldType: 'TEXT',
      fieldTypeAttributes: {
        format: 'UNFORMATTED',
        fieldSize: 100,
      },
      isList: true,
      isRequired: false,
      isUnique: false,
      defaultValue: 'Scalar List Default Value 1',
      relation: null,
    },
    {
      name: 'nestedRelation',
      displayName: 'Nested Relation',
      description: null,
      fieldType: 'RELATION',
      fieldTypeAttributes: null,
      isList: false,
      isRequired: false,
      isUnique: null,
      defaultValue: null,
      relation: {
        id: 'RELATION_FIELD_ID_5',
        relationTableName: 'RELATION_TABLE_NAME_3',
        relationFieldName: 'aid',
        refTable: {
          id: 'RELATION_TABLE_SCHEMA_ID',
        },
        refFieldIsList: false,
        refFieldIsRequired: true,
      },
    },
    {
      name: 'nestedRelationList',
      displayName: 'Nested Relation List',
      description: null,
      fieldType: 'RELATION',
      fieldTypeAttributes: null,
      isList: true,
      isRequired: false,
      isUnique: null,
      defaultValue: null,
      relation: {
        id: 'RELATION_FIELD_ID_6',
        relationTableName: 'RELATION_TABLE_NAME_4',
        relationFieldName: 'aid',
        refTable: {
          id: 'RELATION_TABLE_SCHEMA_ID',
        },
        refFieldIsList: false,
        refFieldIsRequired: true,
      },
    },
  ],
}];

describe('As developer, I can format for create mutation,', () => {
  it('Flat data.', () => {
    const data = {
      scalar: 'Scalar Value',
      scalarList: ['Scalar List Value'],
    };

    expect(formatDataForMutation(MUTATION_TYPE.CREATE, 'tableSchema', data, SCHEMA)).toEqual(data);
  });

  it('Data with relation reference.', () => {
    const data = {
      relation: '5b32159b66a4500f96285626',
    };

    expect(formatDataForMutation(MUTATION_TYPE.CREATE, 'tableSchema', data, SCHEMA)).toEqual({
      relation: {
        connect: { id: '5b32159b66a4500f96285626' },
      },
    });
  });

  it('Data with relation.', () => {
    const data = {
      relation: {
        scalar: 'Relation Scalar Value',
        scalarList: [
          'Relation Scalar List Value',
        ],
      },
    };

    expect(formatDataForMutation(MUTATION_TYPE.CREATE, 'tableSchema', data, SCHEMA)).toEqual({
      relation: {
        create: {
          scalar: 'Relation Scalar Value',
          scalarList: [
            'Relation Scalar List Value',
          ],
        },
      },
    });
  });

  it('Data with file.', () => {
    const file = new File([''], 'filename');

    const data = {
      file,
    };

    expect(formatDataForMutation(MUTATION_TYPE.CREATE, 'tableSchema', data, SCHEMA)).toEqual({
      file: {
        create: {
          $file: file,
        },
      },
    });
  });

  it('Data with relation list reference.', () => {
    const data = {
      relationList: [
        '5b32159b66a450c047285628',
        '5b32159b66a450fae928562a',
      ],
    };

    expect(formatDataForMutation(MUTATION_TYPE.CREATE, 'tableSchema', data, SCHEMA)).toEqual({
      relationList: {
        connect: [
          { id: '5b32159b66a450c047285628' },
          { id: '5b32159b66a450fae928562a' },
        ],
      },
    });
  });

  it('Data with relation list.', () => {
    const data = {
      relationList: [{
        scalar: 'Relation List Scalar Value',
        scalarList: [
          'Relation List Scalar List Value',
        ],
      }],
    };

    expect(formatDataForMutation(MUTATION_TYPE.CREATE, 'tableSchema', data, SCHEMA)).toEqual({
      relationList: {
        create: [{
          scalar: 'Relation List Scalar Value',
          scalarList: [
            'Relation List Scalar List Value',
          ],
        }],
      },
    });
  });

  it('Data with file list.', () => {
    const file = new File([''], 'filename');

    const data = {
      fileList: [file],
    };

    expect(formatDataForMutation(MUTATION_TYPE.CREATE, 'tableSchema', data, SCHEMA)).toEqual({
      fileList: {
        create: [{
          $file: file,
        }],
      },
    });
  });

  it('Compelex data.', () => {
    const data = {
      scalar: 'Scalar Value',
      scalarList: [
        'Scalar List Value',
      ],
      relation: {
        scalar: 'Relation Scalar Value',
      },
      relationList: [{
        scalar: 'Relation List Scalar Value',
        scalarList: [
          'Relation List Scalar List Value',
        ],
        nestedRelation: '5b32159b66a450c047285628',
        nestedRelationList: [{
          scalar: 'Relation List Nested Relation List Scalar Value',
          scalarList: [
            'Relation List Nested Relation List Scalar List Value',
          ],
        }],
      }],
    };

    expect(formatDataForMutation(MUTATION_TYPE.CREATE, 'tableSchema', data, SCHEMA)).toMatchSnapshot();
  });
});

describe('As developer, I can format for update mutation,', () => {
  it('Flat data.', () => {
    const data = {
      scalar: 'Scalar Value',
      scalarList: ['Scalar List Value'],
    };

    expect(formatDataForMutation(MUTATION_TYPE.UPDATE, 'tableSchema', data, SCHEMA)).toEqual(data);
  });

  it('Data with relation reference.', () => {
    const data = {
      relation: '5b32159b66a4500f96285626',
    };

    expect(formatDataForMutation(MUTATION_TYPE.UPDATE, 'tableSchema', data, SCHEMA)).toEqual({
      relation: {
        reconnect: { id: '5b32159b66a4500f96285626' },
      },
    });
  });

  it('Data with relation.', () => {
    const data = {
      relation: {
        scalar: 'Relation Scalar Value',
        scalarList: [
          'Relation Scalar List Value',
        ],
      },
    };

    expect(formatDataForMutation(MUTATION_TYPE.UPDATE, 'tableSchema', data, SCHEMA)).toEqual({
      relation: {
        create: {
          scalar: 'Relation Scalar Value',
          scalarList: [
            'Relation Scalar List Value',
          ],
        },
      },
    });
  });

  it('Data with relation list reference.', () => {
    const data = {
      relationList: [
        '5b32159b66a450c047285628',
        '5b32159b66a450fae928562a',
      ],
    };

    expect(formatDataForMutation(MUTATION_TYPE.UPDATE, 'tableSchema', data, SCHEMA)).toEqual({
      relationList: {
        reconnect: [
          { id: '5b32159b66a450c047285628' },
          { id: '5b32159b66a450fae928562a' },
        ],
      },
    });
  });

  it('Data with relation list.', () => {
    const data = {
      relationList: [{
        scalar: 'Relation List Scalar Value',
        scalarList: [
          'Relation List Scalar List Value',
        ],
      }],
    };

    expect(formatDataForMutation(MUTATION_TYPE.UPDATE, 'tableSchema', data, SCHEMA)).toEqual({
      relationList: {
        create: [{
          scalar: 'Relation List Scalar Value',
          scalarList: [
            'Relation List Scalar List Value',
          ],
        }],
      },
    });
  });

  it('Compelex data.', () => {
    const data = {
      scalar: 'Scalar Value',
      scalarList: [
        'Scalar List Value',
      ],
      relation: {
        scalar: 'Relation Scalar Value',
      },
      relationList: [{
        scalar: 'Relation List Scalar Value',
        scalarList: [
          'Relation List Scalar List Value',
        ],
        nestedRelation: '5b32159b66a450c047285628',
        nestedRelationList: [{
          scalar: 'Relation List Nested Relation List Scalar Value',
          scalarList: [
            'Relation List Nested Relation List Scalar List Value',
          ],
        }],
      }],
    };

    expect(formatDataForMutation(MUTATION_TYPE.UPDATE, 'tableSchema', data, SCHEMA)).toMatchSnapshot();
  });
});
