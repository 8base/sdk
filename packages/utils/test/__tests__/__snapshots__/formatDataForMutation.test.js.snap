// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`As developer, I can format for create mutation, Compelex data. 1`] = `
Object {
  "relation": Object {
    "create": Object {
      "scalar": "Relation Scalar Value",
    },
  },
  "relationList": Object {
    "create": Array [
      Object {
        "nestedRelation": Object {
          "connect": Object {
            "id": "5b32159b66a450c047285628",
          },
        },
        "nestedRelationList": Object {
          "create": Array [
            Object {
              "scalar": "Relation List Nested Relation List Scalar Value",
              "scalarList": Array [
                "Relation List Nested Relation List Scalar List Value",
              ],
            },
          ],
        },
        "scalar": "Relation List Scalar Value",
        "scalarList": Array [
          "Relation List Scalar List Value",
        ],
      },
    ],
  },
  "scalar": "Scalar Value",
  "scalarList": Array [
    "Scalar List Value",
  ],
}
`;

exports[`As developer, I can format for update mutation, Compelex data. 1`] = `
Object {
  "relation": Object {
    "create": Object {
      "scalar": "Relation Scalar Value",
    },
  },
  "relationList": Object {
    "create": Array [
      Object {
        "nestedRelation": Object {
          "reconnect": Object {
            "id": "5b32159b66a450c047285628",
          },
        },
        "nestedRelationList": Object {
          "create": Array [
            Object {
              "scalar": "Relation List Nested Relation List Scalar Value",
              "scalarList": Array [
                "Relation List Nested Relation List Scalar List Value",
              ],
            },
          ],
        },
        "scalar": "Relation List Scalar Value",
        "scalarList": Array [
          "Relation List Scalar List Value",
        ],
      },
    ],
  },
  "scalar": "Scalar Value",
  "scalarList": Array [
    "Scalar List Value",
  ],
}
`;
